### ISSUE TITLE
virtual population bug: lean option is broken

### ISSUE THREAD

**Issue by TrejGun** [Automattic/mongoose#4288]
```js
var q = require("q");
var mongoose = require("mongoose");
mongoose.set("debug", true);
mongoose.Promise = q.Promise;

var db = mongoose.createConnection();
db.open("mongodb://localhost/issue-123");


var GroupSchema = new mongoose.Schema({
	name: {
		type: String
	}
}, {
	timestamps: true,
	versionKey: false
});

GroupSchema.virtual("users", {
	ref: "User",
	localField: "_id",
	foreignField: "group"
});

var Group = db.model("Group", GroupSchema);

var UserSchema = new mongoose.Schema({
	group: {
		type: mongoose.Schema.Types.ObjectId,
		ref: "Group"
	},
	name: {
		type: String
	}
}, {
	timestamps: true,
	versionKey: false
});

var User = db.model("User", UserSchema);

Group.create({name: "my group 1"})
	.then(function (group) {
		console.log("group", group);
		return User.create({name: "Trej", group: group._id})
			.then(function (user) {
				console.log("user", user);
				return Group.find({_id: {$in: [group._id]}})
					//.lean(true) // <- uncomment me!
					.populate({path: "users"})
					.then(function (result) {
						console.log("result", result);
						console.log("result[0].user", result[0].users); // <- singular
					});
			});
	});
```

virtual population should return an array in this case and works good with lean=false
but if lean=true it converts array of one element to this element
two and more elements works good too

**Comment by vkarpov15** [comment]
@TrejGun have you tried 4.5.3? I thought this was fixed in #4276

**Comment by TrejGun** [comment]
no this one was not fixed

```
result[0].user { _id: 577fb3bd7e564dfdc148da84,
  updatedAt: 2016-07-08T14:07:57.512Z,
  createdAt: 2016-07-08T14:07:57.512Z,
  name: 'Trej',
  group: 577fb3bd7e564dfdc148da83 }
```
vs
```
result[0].user [ { _id: 577fb3d4f2e2f609c2c94cec,
    updatedAt: 2016-07-08T14:08:20.854Z,
    createdAt: 2016-07-08T14:08:20.854Z,
    name: 'Trej',
    group: 577fb3d4f2e2f609c2c94ceb } ]
```
